pkg/controllers/managementuser/secret/secret.go

```diff
@@ -250,12 +250,14 @@ func getNamespacedSecret(obj *corev1.Secret, namespace string) *corev1.Secret {
 	namespacedSecret.Namespace = namespace
 	namespacedSecret.Type = obj.Type
 	namespacedSecret.Annotations = make(map[string]string)
-	copyMap(obj.Annotations, namespacedSecret.Annotations)
+	namespacedSecret.Labels = make(map[string]string)
```

Comment:
I'm not aware enough about the reasons why we might have intentionally omitted copying the labels in the first place, but from what I can tell this LGTM.

@kinarashah any thoughts here? Since it seems like you made the last change in https://github.com/rancher/rancher/commit/413c663813f6b6bf543891c3fd7cce9700a2e213.

Reply:
@genexpr @aiyengar2 I don't remember why we skipped labels, PR is very old. But we should make sure that labels don't include any keys that controllers check for on sync and run some logic, this could be specific to project level scope and not run for each namespace.

Reply:
Hmm @kinarashah I'm not aware of anything that controllers rely on to sync or run some logic, but I'll leave it to other reviewers to perhaps comment. 

I also think that controllers using labels on secrets to execute some action seems like something we should fix rather than something we should prevent in the code here, since they could create that secret via kubectl with the same permissions anyways.

@genexpr if you can identify any such keys in the code (just search for constants or vars that we define that have to do with secrets), I think it might make sense to start a `denylist` here that deletes those labels before persisting it. But unless we can identify such a case, I think the current PR looks good to me still.

Reply:
@aiyengar2 It's not that the controller would execute some action using labels, it'd be more like what you see in the controller, it relies the projectID from the annotation and if it exists, runs some logic. So it's unrelated to user doing anything. They're rancher keys and not user keys. The idea tho is the same as you mentioned, if in case you find a key is being used for a purpose that isn't required to run for namespacedSecret, we shouldn't save it on this secret. For this, we're on the same page based on your last paragraph and the PR is good otherwise.

Reply:
I looked for it, and we're good.

Reply:
ack, thanks for the explanation! https://github.com/rancher/rancher/pull/34923#discussion_r721765205